Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,344 @@
+    /// </summary>
+    public class TapToPlace : Solver, IMixedRealityPointerHandler
+    {
+        
+        [SerializeField]
+        [Tooltip("The default distance (in meters) an object will be placed relative to the TrackedTargetType forward in the SolverHandler." +
+            " The GameObjectToPlace will be placed at the default placement distance if a surface is not hit by the raycast.")]
+        private float defaultPlacementDistance = 1.5f;
+
+        /// <summary>
+        /// The default distance (in meters) an object will be placed relative to the TrackedTargetType forward in the SolverHandler.
+        /// The GameObjectToPlace will be placed at the default placement distance if a surface is not hit by the raycast.
+        /// </summary>
+        public float DefaultPlacementDistance
+        {
+            get => defaultPlacementDistance;
+            set => defaultPlacementDistance = value;    
+        }
+
+        [SerializeField]
+        [Tooltip("Max distance (in meters) to place an object if there is a raycast hit on a surface.")]
+        private float maxRaycastDistance = 20.0f;
+
+        /// <summary>
+        /// The max distance (in meters) to place an object if there is a raycast hit on a surface
+        /// </summary>
+        public float MaxRaycastDistance
+        {
+            get => maxRaycastDistance;
+            set => maxRaycastDistance = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.")]
+        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.
+        /// </summary>
+        public LayerMask[] MagneticSurfaces
+        {
+            get => magneticSurfaces;
+            set => magneticSurfaces = value;
+        }
+
+        /// <summary>
+        /// Is the gameObjectToPlace currently in the placing state? This state is activated when the GameObjectToPlace
+        /// is selected.
+        /// </summary>
+        public bool IsBeingPlaced { get; protected set; }
+
+        [SerializeField]
+        [Tooltip("If true, the gameObjectToPlace will start in the placing state.  The object will immediately start" +
+        " following the TrackedTargetType (Head or Controller Ray) and then a tap is required to place the object.  AutoStart" +
+         " is only called once in Start();")]
+        private bool autoStart = false;
+
+        /// <summary>
+        /// If true, the gameObjectToPlace will start in the placing state.  The object will immediately start
+        /// following the TrackedTargetType (Head or Controller Ray) and then a tap is required to place the object.  AutoStart
+        /// is only called once in Start();
```

Comment:
Nit: The last sentence here is super interesting, I think that what the comment here should say is about what as a consumer you should be aware of:

"AutoStart is only called once in Start" -> "This value must be modified before Start() is invoked in order to have any effect"

Reply:
i.e. when writing comments for someone using this, it's best to put it in terms or information that's useful/expedient for the consumer.

Reply:
Otherwise, the consumer has the make the mental leap here (AutoStart is only called once in start? Oh that means... I probably need to modify it before Start() is invoked in order for this to be useful)

Reply:
Comment updated